src/app/FakeLib/AzureKudu.fs

```diff
@@ -56,3 +56,43 @@ let kuduSync() =
             (TimeSpan.FromMinutes 5.)
     output |> Seq.iter (fun cm -> printfn "%O: %s" cm.Timestamp cm.Message)
     if not succeeded then failwith "Error occurred during Kudu Sync deployment."
+
+/// Kudu ZipDeploy parameters
+type ZipDeployParams =
+  { /// The url of the website, usually in the format of https://<yourwebsite>.scm.azurewebsites.net
+    Url : Uri
+    /// The WebDeploy or Git username, usually the $username from the site's publish profile
+    UserName : string
+    /// The WebDeploy or Git Password
+    Password : string
+    /// The path to the zip archive to upload
+    PackageLocation: string }
+
+/// Synchronizes contents of the zip package with the target web app using Kudu ZipDeploy.
+/// See https://blogs.msdn.microsoft.com/appserviceteam/2017/10/16/zip-push-deployment-for-web-apps-functions-and-webjobs/
+let zipDeploy { Url = uri; UserName = username; Password = password; PackageLocation = zipFile } =
+    // Create the web request.
+    let request =
+        Net.HttpWebRequest.Create(uri.AbsoluteUri + "api/zipdeploy",
+                                  Method = "POST",
+                                  ContentType = "multipart/form-data",
+                                  Timeout = 300000) :?> Net.HttpWebRequest
+
+    // Set the authorization header.
+    let authToken =
+        Convert.ToBase64String(Text.Encoding.ASCII.GetBytes(sprintf "%s:%s" username password))
+    request.Headers.Add("Authorization", sprintf "Basic %s" authToken)
+
+    // Write the zip file to the request stream, then flush and close it to send.
+    do  use fileStream = new FileStream(zipFile, FileMode.Open)
+        use inFile = request.GetRequestStream()
+        fileStream.CopyTo(inFile)
+        inFile.Flush()
+        inFile.Close()
+
+    // Get the response. If 200 OK, then the deploy succeeded. Otherwise, the deploy failed.
+    use response = request.GetResponse() :?> Net.HttpWebResponse
+    if response.StatusCode = Net.HttpStatusCode.OK then
+        logfn "Deployed %s" uri.AbsoluteUri
```

Comment:
Is it appropriate to log out success? It's possible to request an asynchronous deployment that can be tracked by calling another API to check status. I used this approach for its simplicity and because I don't want to poll for status updates. However, I could understand a desire to switch to the polling model, if that's desired.

Reply:
Please be aware that synchronous zip-deploys return 200 OK even if the deploy fails. I've suggested that they give better feedback on push deploys, but it doesn't look like they are prioritizing it: https://github.com/projectkudu/kudu/pull/2641

You should probably do another call on 200 OK to check if the deploy actually succeeded or not.